src/common/commonutils/FileUtils.c

```diff
@@ -367,6 +367,117 @@ int SetFileAccess(const char* fileName, unsigned int desiredOwnerId, unsigned in
     return result;
 }
 
+int CheckDirectoryAccess(const char* name, unsigned int desiredOwnerId, unsigned int desiredGroupId, unsigned int desiredAccess, bool rootCanOverwriteOwnership, void* log)
+{
+    struct stat statStruct = { 0 };
+    mode_t currentMode = 0;
+    mode_t desiredMode = 0;
+    int result = ENOENT;
+
+    if (NULL == name)
+    {
+        OsConfigLogError(log, "CheckAccess called with an invalid name argument");
+        return EINVAL;
+    }
+
+    if (DirectoryExists(name))
+    {
+        if (0 == (result = stat(name, &statStruct)))
+        {
+            if ((((uid_t)desiredOwnerId == statStruct.st_uid) && ((gid_t)desiredGroupId == statStruct.st_gid)) ||
+                (rootCanOverwriteOwnership && (0 == statStruct.st_uid) && (0 == statStruct.st_gid)))
+            {
+                currentMode = FilterFileAccessFlags(statStruct.st_mode);
+                desiredMode = FilterFileAccessFlags(desiredAccess);
+
+                if ((((desiredMode & S_IRWXU) == (currentMode & S_IRWXU)) || (0 == (desiredMode & S_IRWXU))) &&
+                    (((desiredMode & S_IRWXG) == (currentMode & S_IRWXG)) || (0 == (desiredMode & S_IRWXG))) &&
+                    (((desiredMode & S_IRWXO) == (currentMode & S_IRWXO)) || (0 == (desiredMode & S_IRWXO))))
+                {
+                    OsConfigLogInfo(log, "CheckDirectoryAccess: access to '%s' (%u, %u, %u-%u) matches expected (%u, %u, %u-%u)",
+                        name, statStruct.st_uid, statStruct.st_gid, statStruct.st_mode, currentMode,
+                        desiredOwnerId, desiredGroupId, desiredAccess, desiredMode);
+                    result = 0;
+                }
+                else
+                {
+                    OsConfigLogError(log, "CheckDirectoryAccess: no matching access permissions for '%s' (%u-%u) versus expected (%u-%u)",

```

Comment:
nit: Here and elsewhere, it may be clearer to just say _"expected"_ instead of _"versus expected"_